src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -197,4 +233,17 @@ else if(purpose.equals("make public")){
     }
     conversationStore.updateConversation(conversation);
   }
+
+  /*
+  * This function splits source by whitespace and checks if substring 
+  * is contained as a standalone word, ignore case.
+  */
+  private boolean containsExactMatch(String source, String substring) {
+    for (String word : source.split("\\s+")) {
```

Comment:
Note punctuation won't work with this implementation. Something like `@bot: something` or `hi @bot!` won't work.
Maybe we should stick with assuming the mention will be the first word? This is simpler, and it's also a reasonable restriction. If you still want this flexibility, you'll need to also split on punctuation characters. See this answer: https://stackoverflow.com/a/7385150/1768702.